features/location/api/build.gradle.kts

```diff
@@ -14,14 +14,33 @@
  * limitations under the License.
  */
 
+import java.util.Properties
+
 plugins {
     id("io.element.android-compose-library")
     alias(libs.plugins.ksp)
     id("kotlin-parcelize")
 }
 
+fun readLocalProperty(name: String) = Properties().apply {
+    try {
+        load(rootProject.file("local.properties").reader())
+    } catch (ignored: java.io.IOException) {
+    }
+}[name]
+
 android {
     namespace = "io.element.android.features.location.api"
+
+    defaultConfig {
+        resValue(
+            type = "string",
+            name = "maptiler_api_key",
+            value = System.getenv("ELEMENT_ANDROID_MAPTILER_API_KEY")
+                ?: readLocalProperty("services.maptiler.apikey") as? String
```

Comment:
I did not test it, but can't you just invoke `project.property("services.maptiler.apikey")`(and remove the `fun readLocalProperty`), so that the key can be placed in any project property file, or even passed in the command line when building the project.

Reply:
That's what I did originally, but gradle doesn't actually include configuration from `local.properties` (or any equivalent) in its normal property resolution. The only file it sources from that wouldn't be committed is `$GRALDE_USER_HOME/gradle.properties`, and putting project-specific secrets in a global file seems a bit gross (it gets very messy if every project does that, risks having name collisions, leaking secrets to other builds, etc).

We could use a plugin like [gradle-properties-plugin](https://github.com/stevesaliman/gradle-properties-plugin), but that felt a bit overkill for now.

Reply:
OK, thanks. TIL gradle ignores `local.properties`. Related: https://github.com/gradle/gradle/issues/12283

Reply:
@csmith why you used `resValue` and not `buildConfigField` ?

Reply:
> why you used resValue and not buildConfigField ?

No compelling reason. Could have used either, really. I dislike BuildConfigs a little because Android Studio seems incapable of finding them half the time, and they can end up being a dumping ground for any random thing related to the app.